pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -1437,6 +1449,8 @@ cdef class PyObjectHashTable(HashTable):
                 idx = self.table.vals[k]
                 labels[i] = idx
 
+        if return_inverse and return_labels_only:
+            return labels.base, seen_na  # .base -> underlying ndarray
```

Comment:
So the second argument returned by this is either a bool or an ndarray right? Instead of having dynamic return types like that I think it would be better to just pass in a reference to `seen_na` - the caller can choose to ignore the result altogether if they'd like. But that way you don't need a `return_labels_only` argument and can be consistent in what is returned

Reply:
Yeah this was my first idea as well, but it isn't a good idea. We drop the result but we still set an internal variable that signals that there is an external view created, which triggers a copy later in the process because we call into the same hashtable again. Avoiding this copy is part of the performance improvement, so we can't use the same signature.

Reply:
That doesn't matter, but returning the same result as when ``return_labels_only=False`` matters